src/main/java/ludum/dare/screens/UpgradeScreen.java

```diff
@@ -20,38 +24,71 @@
     List<UpgradeGroup> groups = new ArrayList<>();
 
     private int spacePerGroup;
+    private RacerGame game;
+    private List<Player> players;
+
+    public UpgradeScreen(RacerGame game, List<Player> players) {
+        this.game = game;
+        this.players = players;
+        if (MASTER_LIST.size() == 0) {
+            MASTER_LIST.add(new UpgradeOption(DoubleJumpComponent.class, "doubleJump"));
+            MASTER_LIST.add(new UpgradeOption(JetPackComponent.class, "float"));
+            MASTER_LIST.add(new UpgradeOption(MetalComponent.class, "metal"));
+            MASTER_LIST.add(new UpgradeOption(MysteryBagComponent.class, "mystery"));
+            MASTER_LIST.add(new UpgradeOption(SpeedComponent.class, "speed"));
+            MASTER_LIST.add(new UpgradeOption(WallJumpComponent.class, "wallJump"));
+        }
+    }
 
     @Override
     public void show() {
-        camera.position.set(camera.viewportWidth/2, camera.viewportHeight/2, 0);
+        camera.position.set(camera.viewportWidth / 2, camera.viewportHeight / 2, 0);
         camera.update();
         batch = new AnimagicSpriteBatch(camera);
         batch.isShaderOn(false);
 
-        // need a way to get the players in the game.
-        List<GameObject> players = new ArrayList<>();
-        players.add(new GameObject());
-        players.add(new GameObject());
-        players.add(new GameObject());
-        players.add(new GameObject());
-
         spacePerGroup = Gdx.graphics.getHeight() / players.size();
-        for (GameObject player : players) {
+        for (Player player : players) {
             UpgradeGroup upgradeGroup = buildPlayerOptions(player);
             groups.add(upgradeGroup);
         }
     }
 
-    private UpgradeGroup buildPlayerOptions(GameObject player) {
+    private UpgradeGroup buildPlayerOptions(Player player) {
         UpgradeGroup group = new UpgradeGroup();
         group.initialize(player);
+        int tries = 10;
```

Comment:
Maybe just do this infinitely to prevent edge cases? I guess theres the chance that a player will have all upgrades bought except 2.